### PR TITLE
Review comments

### PR DESCRIPTION
**Pull request by Qeas**
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/VladimirMarkelov%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23issuecomment-234077442%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-07-20T20%3A47%3A22Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%202220b37a8e5626fec0646afe53744272c2dd415c%20Makefile%2036%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67548658%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20am%20not%20sure%20if%20it%20is%20required%20here.%20If%20a%20user%20has%20the%20Makefile%20it%20means%20the%20user%20has%20already%20done%20%27git%20clone%27.%20So%20he%20does%20not%20need%20to%20run%20setup%20to%20do%20%27go%20get%27%20and%20download%20the%20sources.%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A44%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%2C%20%7B%22body%22%3A%20%22this%20will%20build%20the%20dependencies%20too%22%2C%20%22created_at%22%3A%20%222016-07-20T20%3A29%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5867897%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Qeas%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile%3AL1-18%22%7D%2C%20%22Pull%202220b37a8e5626fec0646afe53744272c2dd415c%20README.md%2045%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550638%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22If%20%27go%20get%27%20installs%20all%20the%20required%20modules%2C%20why%20did%20you%20replace%20%27go%20get%27%20with%20%27git%20clone%27%3F%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A57%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%2C%20%7B%22body%22%3A%20%22well%20the%20user%20basicly%20needs%20only%20the%20Makefile%2C%20which%20will%20use%20get%20afterwards%22%2C%20%22created_at%22%3A%20%222016-07-20T20%3A46%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5867897%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Qeas%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20README.md%3AL53-63%22%7D%2C%20%22Pull%202220b37a8e5626fec0646afe53744272c2dd415c%20README.md%2054%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550484%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22On%20a%20user%20computer%20%24HOME/gocode%20most%20likely%20does%20not%20exist.%20I%20do%20not%20remember%20if%20golang%20creates%20the%20folder%20automatically%20while%20running%20%27go%20get%27%20or%20it%20wants%20it%20to%20be%20pre-created.%20In%20the%20latter%20case%2C%20you%20should%20warn%20user%20to%20create%20the%20directory%20with%20mkdir%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A56%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20README.md%3AL68-76%22%7D%2C%20%22Pull%202220b37a8e5626fec0646afe53744272c2dd415c%20README.md%2046%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550043%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Did%20you%20test%20it%3F%20You%20removed%20section%20%27all%27%20from%20Makefile%2C%20so%20bare%20%27make%27%20may%20not%20work.%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A53%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%2C%20%7B%22body%22%3A%20%22yes%20it%20starts%20setup%20by%20default%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A57%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5867897%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Qeas%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20README.md%3AL53-63%22%7D%2C%20%22Commit%20976152de0bebb671cf6c92838a1d581965a5ae54%20nvd/nvdapi/nvdapi.go%20117%20157%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54%23commitcomment-18329140%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22double%20%25s%22%2C%20%22created_at%22%3A%20%222016-07-20T21%3A27%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20nvd/nvdapi/nvdapi.go%3AL157%20%28976152d%29%22%7D%2C%20%22Pull%202220b37a8e5626fec0646afe53744272c2dd415c%20Makefile%2042%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67548667%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22The%20script%20is%20run%20inside%20project%20folder%2C%20so%2C%20I%20think%20%27%24GOPATH/src/github.com/nexenta/nexenta-docker-driver%27%20can%20be%20replaced%20with%20%27.%27%20or%20alike.%20It%20applies%20to%20other%20lines%20in%20Makefile%20that%20uses%20full%20path%20instead%20of%20just%20command.%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A44%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%2C%20%7B%22body%22%3A%20%22just%20feels%20more%20secure%20this%20way%2C%20you%20never%20know%20where%20users%20will%20execute%20the%20script%2C%20don%27t%20you%20agree%3F%20%3A%29%22%2C%20%22created_at%22%3A%20%222016-07-20T20%3A30%3A15Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5867897%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Qeas%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile%3AL1-18%22%7D%2C%20%22Pull%202220b37a8e5626fec0646afe53744272c2dd415c%20Makefile%2041%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67548860%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Trying%20to%20download%20golint%20every%20time%3F%20Doesn%27t%20lint%20is%20the%20part%20of%20golang%20package%3F%20If%20not%2C%20I%20think%2C%20it%20is%20better%20to%20place%20%27go%20get%20...%27%20to%20setup%20section%20-%20it%20should%20be%20run%20only%20once%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A45%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%2C%20%7B%22body%22%3A%20%22without%20-u%20flag%20it%20does%20not%20download%20everytime%2C%20it%20checks%20local%20first%20and%20if%20it%20exists%20-%20uses%20local%20version.%20%22%2C%20%22created_at%22%3A%20%222016-07-20T20%3A28%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5867897%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Qeas%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile%3AL1-18%22%7D%2C%20%22Pull%202220b37a8e5626fec0646afe53744272c2dd415c%20Makefile%2057%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67549106%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22If%20the%20script%20is%20running%20in%20the%20project%20directory%20it%20may%20be%20sufficient%20to%20just%20do%20%27go%20clean%27%20without%20adding%20package%20path%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A47%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Makefile%3AL1-18%22%7D%2C%20%22Pull%202220b37a8e5626fec0646afe53744272c2dd415c%20README.md%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67549783%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20GCC%20is%20required%3F%20%5Cr%5CnDo%20a%20user%20need%20%27make%27%3F%20Make%20is%20not%20part%20of%20GCC%2C%20in%20Ubuntu%20it%20is%20in%20the%20package%20%27build-essential%27%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A52%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%2C%20%7B%22body%22%3A%20%22on%20a%20clean%20ubuntu%20install%20one%20of%20the%20dependencies%20did%20not%20work%20without%20gcc%5Cr%5Cndon%27t%20remember%20which%20one%20though%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A56%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5867897%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Qeas%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20README.md%3AL19-51%22%7D%2C%20%22Commit%20976152de0bebb671cf6c92838a1d581965a5ae54%20nvd/nvdapi/nvdapi.go%20129%20169%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54%23commitcomment-18329193%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22all%20errors%20inside%20the%20function%20are%20logged%20except%20this%20one%22%2C%20%22created_at%22%3A%20%222016-07-20T21%3A32%3A39Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20nvd/nvdapi/nvdapi.go%3AL169%20%28976152d%29%22%7D%2C%20%22Commit%20976152de0bebb671cf6c92838a1d581965a5ae54%20nvd/nvdapi/nvdapi.go%2092%20125%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54%23commitcomment-18329039%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20double%20%25s%3F%22%2C%20%22created_at%22%3A%20%222016-07-20T21%3A20%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%2C%20%7B%22body%22%3A%20%22double%20%25s%22%2C%20%22created_at%22%3A%20%222016-07-20T21%3A27%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20nvd/nvdapi/nvdapi.go%3AL125%20%28976152d%29%22%7D%2C%20%22Pull%202220b37a8e5626fec0646afe53744272c2dd415c%20nvd/nvdapi/nvdapi.go%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550909%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22For%20the%20future%3F%20%3A%29%20Port%20is%2016-bit%20integer.%20So%2C%20int16%20or%20int32%20is%20enough%22%2C%20%22created_at%22%3A%20%222016-06-17T17%3A58%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1577186%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/VladimirMarkelov%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20nvd/nvdapi/nvdapi.go%3AL7-73%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23issuecomment-234077442%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67548658%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67548667%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67548860%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67549106%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67549783%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550043%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550484%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550528%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550614%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550638%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r67550909%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r71600331%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r71600529%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r71600679%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/pull/2%23discussion_r71603381%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54%23commitcomment-18329039%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54%23commitcomment-18329137%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54%23commitcomment-18329140%22%2C%20%22https%3A//github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54%23commitcomment-18329193%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/VladimirMarkelov'></a>

- [ ] <a href='#crh-comment-Pull 2220b37a8e5626fec0646afe53744272c2dd415c Makefile 36'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2#discussion_r67548658'>File: Makefile:L1-18</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> I am not sure if it is required here. If a user has the Makefile it means the user has already done 'git clone'. So he does not need to run setup to do 'go get' and download the sources.
- <a href='https://github.com/Qeas'></a> this will build the dependencies too
- [ ] <a href='#crh-comment-Pull 2220b37a8e5626fec0646afe53744272c2dd415c Makefile 42'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2#discussion_r67548667'>File: Makefile:L1-18</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> The script is run inside project folder, so, I think '$GOPATH/src/github.com/nexenta/nexenta-docker-driver' can be replaced with '.' or alike. It applies to other lines in Makefile that uses full path instead of just command.
- <a href='https://github.com/Qeas'></a> just feels more secure this way, you never know where users will execute the script, don't you agree? :)
- [ ] <a href='#crh-comment-Pull 2220b37a8e5626fec0646afe53744272c2dd415c Makefile 41'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2#discussion_r67548860'>File: Makefile:L1-18</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> Trying to download golint every time? Doesn't lint is the part of golang package? If not, I think, it is better to place 'go get ...' to setup section - it should be run only once
- <a href='https://github.com/Qeas'></a> without -u flag it does not download everytime, it checks local first and if it exists - uses local version.
- [ ] <a href='#crh-comment-Pull 2220b37a8e5626fec0646afe53744272c2dd415c Makefile 57'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2#discussion_r67549106'>File: Makefile:L1-18</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> If the script is running in the project directory it may be sufficient to just do 'go clean' without adding package path
- [ ] <a href='#crh-comment-Pull 2220b37a8e5626fec0646afe53744272c2dd415c README.md 4'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2#discussion_r67549783'>File: README.md:L19-51</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> Why GCC is required?
Do a user need 'make'? Make is not part of GCC, in Ubuntu it is in the package 'build-essential'
- <a href='https://github.com/Qeas'></a> on a clean ubuntu install one of the dependencies did not work without gcc
don't remember which one though
- [ ] <a href='#crh-comment-Pull 2220b37a8e5626fec0646afe53744272c2dd415c README.md 46'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2#discussion_r67550043'>File: README.md:L53-63</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> Did you test it? You removed section 'all' from Makefile, so bare 'make' may not work.
- <a href='https://github.com/Qeas'></a> yes it starts setup by default
- [ ] <a href='#crh-comment-Pull 2220b37a8e5626fec0646afe53744272c2dd415c README.md 54'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2#discussion_r67550484'>File: README.md:L68-76</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> On a user computer $HOME/gocode most likely does not exist. I do not remember if golang creates the folder automatically while running 'go get' or it wants it to be pre-created. In the latter case, you should warn user to create the directory with mkdir
- [ ] <a href='#crh-comment-Pull 2220b37a8e5626fec0646afe53744272c2dd415c README.md 45'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2#discussion_r67550638'>File: README.md:L53-63</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> If 'go get' installs all the required modules, why did you replace 'go get' with 'git clone'? :)
- <a href='https://github.com/Qeas'></a> well the user basicly needs only the Makefile, which will use get afterwards
- [ ] <a href='#crh-comment-Pull 2220b37a8e5626fec0646afe53744272c2dd415c nvd/nvdapi/nvdapi.go 21'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2#discussion_r67550909'>File: nvd/nvdapi/nvdapi.go:L7-73</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> For the future? :) Port is 16-bit integer. So, int16 or int32 is enough
- [ ] <a href='#crh-comment-Commit 976152de0bebb671cf6c92838a1d581965a5ae54 nvd/nvdapi/nvdapi.go 92 125'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54#commitcomment-18329039'>File: nvd/nvdapi/nvdapi.go:L125 (976152d)</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> Why double %s?
- <a href='https://github.com/VladimirMarkelov'></a> double %s
- [ ] <a href='#crh-comment-Commit 976152de0bebb671cf6c92838a1d581965a5ae54 nvd/nvdapi/nvdapi.go 117 157'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54#commitcomment-18329140'>File: nvd/nvdapi/nvdapi.go:L157 (976152d)</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> double %s
- [ ] <a href='#crh-comment-Commit 976152de0bebb671cf6c92838a1d581965a5ae54 nvd/nvdapi/nvdapi.go 129 169'></a> &nbsp;<b><a href='https://github.com/Nexenta/nexenta-docker-driver/commit/976152de0bebb671cf6c92838a1d581965a5ae54#commitcomment-18329193'>File: nvd/nvdapi/nvdapi.go:L169 (976152d)</a></b>
- <a href='https://github.com/VladimirMarkelov'></a> all errors inside the function are logged except this one

<a href='https://www.codereviewhub.com/Nexenta/nexenta-docker-driver/pull/2?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/Nexenta/nexenta-docker-driver/pull/2?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/Nexenta/nexenta-docker-driver/pull/2?approve=0'></a>&nbsp;<a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by n3m8tz** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/Nexenta/nexenta-docker-driver/pull/2'>Refresh</a>.<a href='#crh-update-none'></a>

File: Makefile [hunk 1]
```diff
@@ -1,51 +1,17 @@
...
-
-build:
-	GOPATH=$(shell pwd) go build $(FLAGS) github.com/Nexenta/nedge-docker-volume/nedv
 
 setup: 
-	mkdir -p src/github.com/Nexenta/nedge-docker-volume/ 
-	cp -R ned/ src/github.com/Nexenta/nedge-docker-volume/nedv 
+	sudo mkdir -p /etc/nvd/
+	sudo cp nvd.json /etc/nvd/
+	go get -u -v github.com/nexenta/nexenta-docker-driver/...
```
Comment (VladimirMarkelov) [inline Makefile]: I am not sure if it is required here. If a user has the Makefile it means the user has already done 'git clone'. So he does not need to run setup to do 'go get' and download the sources.

File: Makefile [hunk 2]
```diff
@@ -1,51 +1,17 @@
...
 
 lint:
-	GOPATH=$(shell pwd) go get -v github.com/golang/lint/golint
-	for file in $$(find . -name '*.go' | grep -v vendor | grep -v '\.pb\.go' | grep -v '\.pb\.gw\.go'); do \
+	go get -v github.com/golang/lint/golint
+	for file in $$(find $GOPATH/src/github.com/nexenta/nexenta-docker-driver -name '*.go' | grep -v vendor | grep -v '\.pb\.go' | grep -v '\.pb\.gw\.go'); do \
```
Comment (VladimirMarkelov) [inline Makefile]: The script is run inside project folder, so, I think '$GOPATH/src/github.com/nexenta/nexenta-docker-driver' can be replaced with '.' or alike.

File: Makefile [hunk 3]
```diff
@@ -1,51 +1,17 @@
...
-	mkdir -p src/github.com/Nexenta/nedge-docker-volume/ 
-	cp -R ned/ src/github.com/Nexenta/nedge-docker-volume/nedv 
+	sudo mkdir -p /etc/nvd/
+	sudo cp nvd.json /etc/nvd/
+	go get -u -v github.com/nexenta/nexenta-docker-driver/...
 
 lint:
-	GOPATH=$(shell pwd) go get -v github.com/golang/lint/golint
-	for file in $$(find . -name '*.go' | grep -v vendor | grep -v '\.pb\.go' | grep -v '\.pb\.gw\.go'); do \
+	go get -v github.com/golang/lint/golint
```
Comment (VladimirMarkelov) [inline Makefile]: Trying to download golint every time? Doesn't lint is the part of golang package? If not, I think, it is better to place 'go get ...' to setup section - it should be run only once

File: Makefile [hunk 4]
```diff
@@ -1,51 +1,17 @@
...
 
 clean:
-	GOPATH=$(shell pwd) go clean
-
-
-clobber:
-	rm -rf src/github.com/Nexenta/nedge-docker-volume
-	rm -rf bin/ pkg/
-
+	go clean github.com/nexenta/nexenta-docker-driver
```
Comment (VladimirMarkelov) [inline Makefile]: If the script is running in the project directory it may be sufficient to just do 'go clean'

File: README.md [hunk 5]
```diff
@@ -19,19 +19,32 @@ environment.  After downloading the appropriate package be sure to scroll down
 to the Linux, Mac OS X, and FreeBSD tarballs section and set up your Go
 environment as per the instructions.
 
+### GCC
```
Comment (VladimirMarkelov) [inline README.md]: Why GCC is required? 
Do a user need 'make'? Make is not part of GCC, in Ubuntu it is in the package 'build-essential'

File: README.md [hunk 6]
```diff
@@ -40,9 +53,10 @@ Default path to config file is
   ```
 
 ## Driver Installation
-After the above Prerequisites are met, Use the standard golang install process:
+After the above Prerequisites are met, clone repository and use the Makefile:
   ```
-  go get -u github.com/nexenta/nexenta-docker-driver/...
+  git clone https://github.com/nexenta/nexenta-docker-driver
+  make
```
Comment (VladimirMarkelov) [inline README.md]: Did you test it? You removed section 'all' from Makefile, so bare 'make' may not work.

File: README.md [hunk 7]
```diff
@@ -54,6 +68,8 @@ either run visudo and edit, or provide an alias in your .bashrc file.
 
 For example in your .bashrc set the following alias after setting up PATH:
   ```
+  export GOPATH=$HOME/gocode
```
Comment (VladimirMarkelov) [inline README.md]: On a user computer $HOME/gocode most likely does not exist. I do not remember if golang creates the folder automatically while running 'go get' or it wants it to be pre-created. In the latter case, you should warn user to create the directory with mkdir

File: README.md (hunk 5 above)
Comment (Qeas) [inline README.md]: on a clean ubuntu install one of the dependencies did not work without gcc

File: README.md (hunk 6 above)
Comment (Qeas) [inline README.md]: yes it starts setup by default

File: README.md [hunk 8]
```diff
@@ -40,9 +53,10 @@ Default path to config file is
   ```
 
 ## Driver Installation
-After the above Prerequisites are met, Use the standard golang install process:
+After the above Prerequisites are met, clone repository and use the Makefile:
   ```
-  go get -u github.com/nexenta/nexenta-docker-driver/...
+  git clone https://github.com/nexenta/nexenta-docker-driver
```
Comment (VladimirMarkelov) [inline README.md]: If 'go get' installs all the required modules, why did you replace 'go get' with 'git clone'? :)

File: nvd/nvdapi/nvdapi.go [hunk 9]
```diff
@@ -7,56 +7,66 @@ import (
...
+const defaultPort int64 = 8443;
 
 type Client struct {
 	Protocol          string
 	Endpoint          string
 	Path              string
 	DefaultVolSize    int64 //bytes
 	Config            *Config
-	Port 			  string
+	Port 			  int64
```
Comment (VladimirMarkelov) [inline nvd/nvdapi/nvdapi.go]: For the future? :) Port is 16-bit integer. So, int16 or int32 is enough

File: Makefile (hunk 3 above)
Comment (Qeas) [inline Makefile]: without -u flag it does not download everytime, it checks local first and if it exists - uses local version.

File: Makefile (hunk 1 above)
Comment (Qeas) [inline Makefile]: this will build the dependencies too

File: Makefile (hunk 2 above)
Comment (Qeas) [inline Makefile]: just feels more secure this way, you never know where users will execute the script, don't you agree? :)

File: README.md (hunk 8 above)
Comment (Qeas) [inline README.md]: well the user basicly needs only the Makefile, which will use get afterwards

**Comment by VladimirMarkelov** [comment]
:shipit:

**Pull request merged**